Dev: remove gulp

ravinderk opened the pull request:
<!-- Make sure to prefix the title with one of New:, Fix:, Changed:, or Security: -->

<!-- Indicate the issue(s) resolved by this PR. -->

Resolves #

## Description

<!-- Summarize the related issue, explain HOW this PR solves the problem, and WHY you made the choices you made. -->
This pull request has the following changes:
- Remove gulp which is used to generate pot but we are using wp cli in the build process.
- Remove empty script

## Affects

<!-- Mention any existing functionality affected by this PR to help inform the reviewer(s). -->
This pull request should not affect any area because we remove unused code.

## Testing Instructions

<!-- Help others test your PR as efficiently as possible. -->
- [ ] Plugin should pass the build process
- [ ] Plugin should not cause errors when activating deactivating or editing admin settings.

## Pre-review Checklist

<!-- Complete tasks prior to requesting a review. Add to this list, but do not remove the base items. -->

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

JasonTheAdams commented:
@ravinderk Why isn't the gulpfile included in this?

ravinderk commented:
@JasonTheAdams, I added the file. Please check.

JasonTheAdams commented:
If we're removing `npm run build` then we're going to need to update the Github actions to allow for that build step to be optional, and then update the workflows here. Otherwise our build steps will fail.

ravinderk commented:
> If we're removing `npm run build` then we're going to need to update the Github actions to allow for that build step to be optional, and then update the workflows here. Otherwise our build steps will fail.

@JasonTheAdams I removed this by mistake. I added it back.

JasonTheAdams commented:
Yeah, but it does nothing, now. This whole add-on basically doesn't need npm anymore, so I'd rather clean it up properly and simply have the build process skip that step.

ravinderk commented:
> Yeah, but it does nothing, now. This whole add-on basically doesn't need npm anymore, so I'd rather clean it up properly and simply have the build process skip that step.

@JasonTheAdams Make sense. I will make a change to the GitHub action.

ravinderk commented:
> Yeah, but it does nothing, now. This whole add-on basically doesn't need npm anymore, so I'd rather clean it up properly and simply have the build process skip that step.

@JasonTheAdams I found that we already have options to do that: https://github.com/impress-org/givewp-github-actions/blob/00a3c02ed92e290f5401d5676761c53832977181/.github/workflows/generate-zip.yml#L14

I used them in the workflow to skip composer and npm install/build.

kjohnson reviewed (commented):
I confirmed that the workflow was updated to not install NPM packages.
